### PR TITLE
CI: Add initial repo config files.

### PR DESCRIPTION
**Pull request by jwagantall**
Initial configuration files for Magma
ci-management repo.

Signed-off-by: Jessica Wagantall <jwagantall@linuxfoundation.org>

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    rev: v1.26.1
+    hooks:
+      - id: yamllint
+
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 2.1.5
+    hooks:
+      - id: shellcheck
+
+  - repo: https://github.com/btford/write-good
```
Comment (tykeal) [inline .pre-commit-config.yaml:36]: We're not adding documentation here, so we can drop all references to write-good

**Review by tykeal — CHANGES_REQUESTED** [review]
Missing .gitlint file to configure the gitlint pre-commit for catching subject lines and missing and extra check for catching missing DCO

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jwagantall) [reply .pre-commit-config.yaml:36]: Will update in the new patch

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    jenkins-job-builder==3.5.0
+commands =
+    jenkins-jobs -l DEBUG test --recursive -o {toxinidir}/archives/job-configs {toxinidir}/jjb:{toxinidir}/.jjb-test
+
+[testenv:jjb-compare-xml]
+basepython = python3
+deps =
+    jenkins-job-builder
+commands =
+    ./jjb-compare-xml.sh
```
Comment (tykeal) [inline tox.ini:30]: This tox job is being defined to call a script that isn't in the tree. Have you run tox locally on this?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+envlist =
+  docs,
+  jjb,
+  jjb-compare-xml,
+  jjb-version,
+  license,
+  pre-commit,
+skipsdist = True
+
+[testenv:docs]
```
Comment (tykeal) [inline tox.ini:12]: I don't think we need the docs job

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,66 @@
+[tox]
+minversion = 1.6
+envlist =
+  docs,
```
Comment (tykeal) [inline tox.ini:4]: I don't think we need the docs job

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+reno~=2.11.2
```
Comment (tykeal) [inline requirements.txt:1]: Reno and sphinx bits aren't needed if we aren't doing docs in the repo. I don't know about the need for yq or not

**Review by tykeal — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 4 above)
Comment (jwagantall) [reply tox.ini:4]: Agreed

File: requirements.txt (hunk 5 above)
Comment (jwagantall) [reply requirements.txt:1]: Agreed

File: tox.ini (hunk 3 above)
Comment (jwagantall) [reply tox.ini:12]: Agreed

File: tox.ini (hunk 2 above)
Comment (jwagantall) [reply tox.ini:30]: jjb-version and jjb will fail for now since I havent brough in the jjb folder.

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+[testenv:pre-commit]
+basepython = python3
+allowlist_externals =
+    /bin/sh
+deps =
+    pre-commit
+passenv = HOME
+commands =
+    pre-commit install --hook-type commit-msg
```
Comment (tykeal) [inline tox.ini:31]: We don't need this line here

**Review by tykeal — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 6 above)
Comment (jwagantall) [reply tox.ini]: changed.. thank you!

**Review by tykeal — APPROVED** [review]

**Pull request merged**